language/move-lang/src/parser/syntax.rs

```diff
@@ -615,15 +597,21 @@ fn parse_term<'input>(tokens: &mut Lexer<'input>) -> Result<Exp, Error> {
             let rhs = parse_call_args(tokens)?;
             Exp_::GlobalCall(n, tys, rhs)
         }
-    });
+
+        _ => {
+            return Err(unexpected_token_error(tokens, "an expression term"));
+        }
+    };
     let end_loc = tokens.previous_end_loc();
     Ok(spanned(tokens.file_name(), start_loc, end_loc, term))
 }
 
 // Parse the subset of expression terms for pack and call operations.
 // This is a helper function for parse_term.
 fn parse_pack_or_call<'input>(tokens: &mut Lexer<'input>) -> Result<Exp_, Error> {
-    let n = parse_module_access(tokens)?;
+    let n = parse_module_access(tokens, || {
+        panic!("parse_pack_or_call with something other than a NameValue or AddressValue token")
```

Comment:
Probably a stupid question: why is this a panic and not an Error?

Reply:
It's not an error that a user should ever see. It's an internal consistency failure. This function is only called in a context where the next token will be either a NameValue or AddressValue token. If that isn't true, then something is badly broken in the compiler.